addons/web/static/src/views/graph/graph_model.js

```diff
@@ -501,9 +501,10 @@ export class GraphModel extends Model {
         const processedGroupBy = [];
         for (const gb of groupBy) {
             const { fieldName, interval } = gb;
-            const { sortable, type } = fields[fieldName];
+            const { sortable, type, store } = fields[fieldName];
             if (
-                !sortable ||
+                // many2many are groupable only when it is stored (cf. odoo/fields.py)
```

Comment:
a many2many is groupable precisely when it is stored (cf. groupable in odoo/fields.py)